src/SFML/Window/WindowImpl.cpp

```diff
@@ -175,30 +176,50 @@ void WindowImpl::setMaximumSize(const std::optional<Vector2u>& maximumSize)
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent(bool block)
+Event WindowImpl::popEventBlocking(Time timeout)
 {
+    const auto startTime = std::chrono::steady_clock::now();
+
+    const auto timedOut = [&]
+    { return timeout != Time::Zero && (std::chrono::steady_clock::now() - startTime) < timeout.toDuration(); };
```

Comment:
If the lambda spreads across multiple lines, can we use:

```suggestion
    {
        return timeout != Time::Zero && (std::chrono::steady_clock::now() - startTime) < timeout.toDuration();
    };
```

Reply:
This is unfortunately up to `clang-format`, not me...

Reply:
Thought so, @ChrisThrasher any option we can tweak? 😄

Reply:
If you feel strongly about the formatting here, I can also manually address it via `// clang-format off`

Reply:
No it's fine for now.